src/uiTest/scala/com/virtuslab/gitmachete/uitest/UITestSuite.scala

```diff
@@ -222,6 +222,20 @@ class UITestSuite extends BaseGitRepositoryBackedIntegrationTestSuite(SETUP_WITH
     project.assertNoUncommittedChanges()
   }
 
+  @Test def squashBranch(): Unit = {
+    project.openGitMacheteTab()
+    project.toolbar.toggleListingCommits()
+    var branchRowsCount = project.refreshModelAndGetRowCount()
+    Assert.assertEquals(18, branchRowsCount)
+    project.checkoutBranch("call-ws")
+    project.contextMenu.openContextMenu("call-ws")
+    project.contextMenu.squash()
+    project.acceptSquash()
+    branchRowsCount = project.refreshModelAndGetRowCount()
+    // call-ws had 3 commits before the squash
+    Assert.assertEquals(16, branchRowsCount)
+  }
+
   private def macheteFilePath: Path = mainGitDirectoryPath.resolve("machete")
 
   private def deleteMacheteFile(): Unit = {
```

Comment:
Wrt. `testCommon/src/test/resources/setup-with-single-remote.sh` &mdash; should the other files be changed to executable? IIRC they're executed as `bash <script>` anyway in our test suites

Reply:
no need for that - that was for local & manual use only

Reply:
Huh 3 files are `chmod +x`ed now... should it stay like that?

Reply:
(they all are `chmod -x` now )